test/pkg.jl

```diff
@@ -141,6 +141,27 @@ temp_pkg_dir() do
     @test endswith(str, string(Pkg.installed("Example")))
     @test isempty(Pkg.dependents("Example"))
 
+
+    #-------
+    # 17364
+    #-------
+    let branch_name = "test-branch",
+        branch_commit = "ba3888212e30a7974ac6803a89e64c7098f4865e"
+
+        # create a branch in Example package
+        LibGit2.with(LibGit2.GitRepo, Pkg.dir("Example")) do repo
+            LibGit2.branch!(repo, branch_name, branch_commit, set_head=false)
+        end
+
+        Pkg.clone(Pkg.dir("Example"), Pkg.dir("Example2"))
+
+        Pkg.checkout("Example2", branch_name)
+
+        LibGit2.with(LibGit2.GitRepo, Pkg.dir("Example2")) do repo
+            @test LibGit2.head_oid(repo) == LibGit2.Oid(branch_commit)
```

Comment:
The test should reproduce the issue that is failing on master. If I run this test, without applying the patch to base, it's supposed to fail otherwise it's not capturing the issue.